ldapdb/backends/ldap/base.py

```diff
@@ -252,13 +259,46 @@ def rename_s(self, dn, newrdn):
         cursor = self._cursor()
         return cursor.connection.rename_s(dn, newrdn)
 
-    def search_s(self, base, scope, filterstr='(objectClass=*)',
-                 attrlist=None):
+    def batched_search_s(self, base, scope, filterstr='(objectClass=*)',
+                         attrlist=None):
```

Comment:
I'd rather keep the old name, and pass the ``batch_size`` as a paremeter to this query.